### ISSUE TITLE
Move to netaddr and bump github.com/mdlayher/ndp

### ISSUE THREAD

**Issue by fedepaol** [metallb/metallb#2376]
### Is your feature request related to a problem?

The ndp package we use moved to use netaddr. It would make sense for us to start using it all over metallb not only because it would allow us to bump ndp but because it's more efficient (more details on https://tailscale.com/blog/netaddr-new-ip-type-for-go ) 

### Describe the solution you'd like

Replacing all the net.Something instance with the corresponding netaddr, add a translation only where an external library accepts net.Something.

### I've read and agree with the following

- [X] I've checked all open and closed issues and my request is not there.
- [X] I've checked all open and closed pull requests and my request is not there.

**Comment by fedepaol** [comment]
> Hi I'd like to take this if possible. I will be looking at the code and at the library and will be back later.

cool! Assigning this to you, if you have any questions reach out on the #metallb-dev slack channel

**Comment by fulviodenza** [comment]
So, I've been giving a look at the library and at the ndp dependency and I have a few considerations:
1. The library linked seems to be deprecated in favor of <a href="https://pkg.go.dev/go4.org/netipx">this library</a>
2. I've noticed that metallb is using a quite old version of ndp, while [the newer versions of this library use net/netip](https://github.com/mdlayher/ndp/commit/3c95c205b9a485f6eeb29632df4c89504a2353dc). Shall we consider using that?
Unless I didn't look in the right, place, it looks to me that we shall consider uysing the net/netip library

Let me know what do you think, thank you!

**Comment by fedepaol** [comment]
> So, I've been giving a look at the library and at the ndp dependency and I have a few considerations:
> 
>     1. The library linked seems to be deprecated in favor of [this library](https://pkg.go.dev/go4.org/netipx)
> 
>     2. I've noticed that metallb is using a quite old version of ndp, while [the newer versions of this library use net/netip](https://github.com/mdlayher/ndp/commit/3c95c205b9a485f6eeb29632df4c89504a2353dc). Shall we consider using that?
>        Unless I didn't look in the right, place, it looks to me that we shall consider uysing the net/netip library
> 
> 
> Let me know what do you think, thank you!

Yes, it was just me looking at the wrong place. I remembered ndp to use the new type in the go stdlib, remembered the post about tailscale (who contributed to the std lib) and went looking for it to find the name of the package (which eventually was the intermediate one).

So, TL;DR: we should use net/netip which comes with the stdlib, sorry again for the confusion :sweat_smile: 

I'll rename the issue.

**Comment by fedepaol** [comment]
@fulviodenza you might find [this article](https://djosephsen.github.io/posts/ipnet/) useful

**Comment by fedepaol** [comment]
@fulviodenza un-assigning this, feel free to ping if / when you want to work on it!
